twistlock/datadog_checks/twistlock/twistlock.py

```diff
@@ -0,0 +1,322 @@
+        service_check_name = self.NAMESPACE + ".can_connect"
+        try:
+            scan_result = self._retrieve_json("/api/v1/hosts")
+            self.service_check(service_check_name, AgentCheck.OK, tags=self.config.tags)
+        except Exception as e:
+            self.warning("cannot retrieve registry data: {}".format(e))
+            self.service_check(service_check_name, AgentCheck.CRITICAL, tags=self.config.tags)
+            return None
+
+        for host in scan_result:
+            if 'hostname' not in host:
+                continue
+
+            hostname = host['hostname']
+            host_tags = ["scanned_host:" + hostname] + self.config.tags
+
+            self._report_service_check(host,
+                                       namespace,
+                                       SCAN_DATE_FORMAT,
+                                       tags=host_tags,
+                                       message="Last scan: " + host.get("scanTime"))
+            self._report_vuln_info(namespace, host, host_tags)
+            self._report_compliance_information(namespace, host, host_tags)
+
+    def report_container_compliance(self):
+        namespace = self.NAMESPACE + ".containers"
+        service_check_name = self.NAMESPACE + ".can_connect"
+        try:
+            scan_result = self._retrieve_json("/api/v1/containers")
+            self.service_check(service_check_name, AgentCheck.OK, tags=self.config.tags)
+        except Exception as e:
+            self.warning("cannot retrieve registry data: {}".format(e))
+            self.service_check(service_check_name, AgentCheck.CRITICAL, tags=self.config.tags)
+            return None
+
+        for container in scan_result:
+            if '_id' not in container:
+                continue
+
+            container_tags = []
+            container_info = container.get('info', {})
+            name = container_info.get('name')
+            if name:
+                container_tags += ["container_name:" + name]
+            image_name = container_info.get('imageName')
+            if image_name:
+                container_tags += ["image_name:" + image_name]
+            container_tags += self.config.tags
+
+            self._report_service_check(container,
+                                       namespace,
+                                       SCAN_DATE_FORMAT,
+                                       tags=container_tags,
+                                       message="Last scan: " + container.get("scanTime"))
+            self._report_compliance_information(namespace, container, container_tags)
+
+    def report_vulnerabilities(self):
+        vuln_containers = self._retrieve_json('/api/v1/stats/vulnerabilities')
+
+        for vuln_container in vuln_containers:
+            host_vulns = vuln_container.get('hostVulnerabilities')
+            image_vulns = vuln_container.get('imageVulnerabilities')
+            if not host_vulns and not image_vulns:
```

Comment:
You can either remove this or both branches below

Reply:
The data structure is dumber than you think it is

Reply:
Can you explain? The boolean logic here is what I speak of